### PR TITLE
Docs: Deploy Jekyll site with GitHub Actions

### LINKED ISSUE #8095: docs: deploy Jekyll with GitHub Actions
## Motivation

Allow Jekyll users to use latest Jekyll or jekyll plugins not supported by GitHub Pages

## Suggestion

Document how to use GitHub Actions to deploy a Jekyll v4 website using jekyll plugins written by the community.

### PR DESCRIPTION
**Pull request by MichaelCurrin**
This is a 🔦 documentation change.

## Summary

Create a instructions for Github Actions CI and added appropriate item to a list on the CI page.

https://deploy-preview-8119--jekyllrb.netlify.com/docs/continuous-integration/github-actions/

## Context

Closes #8095

### CODE REVIEW

**Comment by ashmaroli** [comment]
@MichaelCurrin So this is not a *tutorial*..?
That aside, I see some very minor typos in the text. I could point them out for you in a review, but it'll only clutter the page here. So, I'll leave tips:
- Read through each line slowly as if you were reading through it for the first time. The spelling mistakes are minor enough to be lost on a *quick read through*.
- It is `GitHub` instead of `Github`
- We prefer having <code>```liquid</code> instead of `{% highlight liquid %}` so that the code blocks can be read with highlighting directly on GitHub too.

**Comment by MichaelCurrin** [comment]
Thanks @ashmaroli , I'll make the changes.

The request in the issue was to add this to automation section of Docs. https://github.com/jekyll/jekyll/issues/8095#issuecomment-608875009

But I can move it to tutorials if preferred.

**Comment by ashmaroli** [comment]
> The request in the issue was to add this to automation section of Docs

Oh don't bother then. I forgot about that request.

**Comment by MichaelCurrin** [comment]
I've made requested changes. 🚀

**Comment by MichaelCurrin** [comment]
Using `yaml` formatting for the workflow codeblock shows up as all white. I can use `liquid` but it only highlights `JEKYLL_PAT` in color and not the rest of the workflow.

Is it worth creating an issue to add YAML highlighting?

**Comment by ashmaroli** [comment]
> Using `yaml` formatting for the workflow codeblock shows up as all white.

That is already being handled in `#8079`. So don't worry about it.

**Comment by MichaelCurrin** [comment]
Neat, thanks didn't know about scheduling. Netlify doesn't support scheduling but it does have an endpoint which can be hit to trigger a build.

**Comment by MichaelCurrin** [comment]
@ashmaroli Thanks for the changes. The links also look better - I prefer that style over the index approach I found on another page on the site.

**Comment by ashmaroli** [comment]
@MichaelCurrin You're welcome.
Are you going to push a commit regarding *scheduling*..?

Also, as a reader I'd find it odd that `{{ date | timeago }}` would return something other than `4 years and 3 weeks ago` in the future since it always w.r.t *today* .. Just a personal nit perhaps..

**Comment by MichaelCurrin** [comment]
Oh sure, I will add scheduling part today.

Timeago note makes sense and the info and warning blocks look good.

BTW `[token][]` is not strictly necessary - `[token]` alone will work. I have a working example [here](https://github.com/MichaelCurrin/cheatsheets/blob/master/cheatsheets/markdown.md#link-text).

**Comment by ashmaroli** [comment]
> `[token]` alone will work.

Does that work with kramdown + GFM too..? If yes, great to know!
Thanks.

Also, feel free to make changes as you see fit.

**Comment by MichaelCurrin** [comment]
I'll check on the built page and note here.

**Comment by MichaelCurrin** [comment]
Also, a minor thing, I noticed `margin-left: 2.5px` makes code in note look strangely shifted to the right (in the middle of the line but more noticeable at the start of a line).

Screen Shot 2020-04-16 at 9 38 19 pm

**Comment by ashmaroli** [comment]
@MichaelCurrin I agree that the slight shift to the code does look slightly weird. but let us not bother about CSS issues in this PR.

P.S. Thank you for fixing the typos I missed in the first round of review and consequent changes.

File: docs/_docs/continuous-integration/github-actions.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+the setup simple for the user, one may encounter issues if they also check-in `Gemfile.lock` if it
+was generated with an old version of Bundler.
+
+### Setting up the Action
+
+GitHub Actions are registered for a repository by using a YAML file inside the directory path
+`.github/worksflows` (note the dot at the start). Here we shall employ
+[Jekyll Actions][jekyll-actions] from the Marketplace for its simplicity.
+
+Create a **workflow file**, say `jekyll.yml`, using either the GitHub interface or by pushing
```
Comment (DirtyF) [inline docs/_docs/continuous-integration/github-actions.md:99]:
```suggestion
Create a **workflow file**, say `github-pages.yml`, using either the GitHub interface or by pushing
```

File: docs/_docs/continuous-integration/github-actions.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+field. For example, to build nightly at midnight:
+
+```yaml
+on:
+  schedule:
+    - cron:  '0 0 * * *'
+```
+
+[on.schedule]: https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions#onschedule
+
```
Comment (DirtyF) [inline docs/_docs/continuous-integration/github-actions.md:143]: Most people don't know the `cron` syntax, a link to the  https://crontab.guru/ help would be a nice addition here.

File: docs/_docs/deployment/automated.md [hunk 3]
```diff
@@ -18,6 +18,7 @@ We have guides for the following providers:
 * [Travis CI]({{ '/docs/continuous-integration/travis-ci/' | relative_url }})
 * [CircleCI]({{ '/docs/continuous-integration/circleci/' | relative_url }})
 * [Buddy]({{ '/docs/continuous-integration/buddyworks/' | relative_url }})
+* [GitHub Actions]({{ '/docs/continuous-integration/github-actions/' | relative_url }})
```
Comment (DirtyF) [inline docs/_docs/deployment/automated.md:21]:
```suggestion
* [GitHub Actions]({{ '/docs/continuous-integration/github-actions/' | relative_url }})
* [Travis CI]({{ '/docs/continuous-integration/travis-ci/' | relative_url }})
* [CircleCI]({{ '/docs/continuous-integration/circleci/' | relative_url }})
* [Buddy]({{ '/docs/continuous-integration/buddyworks/' | relative_url }})
```

Let's put it first 😄

File: docs/_docs/continuous-integration/github-actions.md (hunk 1 above)
Comment (DirtyF) [reply docs/_docs/continuous-integration/github-actions.md:99]: or `deploy` whatever conveys best what the action is doing

File: docs/_docs/deployment/automated.md (hunk 3 above)
Comment (MichaelCurrin) [reply docs/_docs/deployment/automated.md:21]: I didn't want to presume, but yes I like that idea :)

**Comment by DirtyF** [comment]
@jekyll: merge +docs

**Pull request merged**